tr/cs-221-states-models.md

```diff
@@ -0,0 +1,980 @@
+
+&#10230; Bu bölümde, s durumunda a eylemini gerçekleştirdiğimizde, Succ(s,a) durumuna varacağımızı varsayıyoruz. Burada amaç, başlangıç durumundan başlayıp bitiş durumuna götüren bir eylem dizisi (a1,a2,a3,a4,...) belirlenmesidir. Bu tür bir problemi çözmek için, amacımız durum-temelli modelleri kullanarak asgari (minimum) maliyet yolunu bulmak olacaktır.
+
+<br>
+
+
+**4. Tree search**
+
+&#10230; Ağaç arama
+
+<br>
+
+
+**5. This category of states-based algorithms explores all possible states and actions. It is quite memory efficient, and is suitable for huge state spaces but the runtime can become exponential in the worst cases.**
+
+&#10230; Bu durum-temelli algoritmalar, olası bütün durum ve eylemleri araştırırlar. Oldukça bellek verimli ve büyük durum uzayları için uygundurlar ancak çalışma zamanı en kötü durumlarda üstel olabilir.
+
+<br>
+
+
+**6. [Self-loop, More than a parent, Cycle, More than a root, Valid tree]**
+
+&#10230; [Kendinden-Döngü, Bir ebeveynden daha fazlası, Çevrim, Bir kökten daha fazlası, Geçerli ağaç]
+
+<br>
+
+
+**7. [Search problem ― A search problem is defined with:, a starting state sstart, possible actions Actions(s) from state s, action cost Cost(s,a) from state s with action a, successor Succ(s,a) of state s after action a, whether an end state was reached IsEnd(s)]**
+
+&#10230; [Arama problemi ― Bir arama problemi aşağıdaki şekilde tanımlanmaktadır:, bir başlangıç durumu sstart, s durumunda gerçekleşebilecek olası eylemler Actions(s), s durumunda gerçekleşen a eyleminin eylem maliyeti Cost(s,a), a eyleminden sonraki varılacak durum Succ(s,a), son duruma ulaşılıp ulaşılamadığı IsEnd(s)]
+
+<br>
+
+
+**8. The objective is to find a path that minimizes the cost.**
+
+&#10230; Amaç, maliyeti en aza indiren bir yol bulmaktır.
+
+<br>
+
+
+**9. Backtracking search ― Backtracking search is a naive recursive algorithm that tries all possibilities to find the minimum cost path. Here, action costs can be either positive or negative.**
+
+&#10230; Geri izleme araması ― Geri izleme araması, asgari (minimum) maliyet yolunu bulmak için tüm olasılıkları deneyen saf bir özyinelemeli algoritmadır. Burada, eylem maliyetleri pozitif ya da negatif olabilir.
+
+<br>
+
+
+**10. Breadth-first search (BFS) ― Breadth-first search is a graph search algorithm that does a level-by-level traversal. We can implement it iteratively with the help of a queue that stores at each step future nodes to be visited. For this algorithm, we can assume action costs to be equal to a constant c⩾0.**
+
+&#10230; Genişlik-ilk arama (BFS) ― Genişlik-ilk arama, seviye seviye arama yapan bir çizge arama algoritmasıdır. Gelecekte her adımda ziyaret edilecek düğümleri tutan bir kuyruk yardımıyla yinelemeli olarak gerçekleyebiliriz. Bu algoritma için, eylem maliyetlerinin belirli bir sabite c⩾0 eşit olduğunu kabul edebiliriz.
```

Comment:
Breadth-first search için 'Genişlik öncelikli arama (Breadth-first search-BFS)" denebilir. Yine kalanında da genişlik öncelikli diye kullanmak daha iyi olur diye düşünüyorum.

Reply:
Hepsini düzelttim.